src/client.ts

```diff
@@ -75,26 +45,14 @@ export class APIClient {
    * @param iteratee receives each resource to produce entities/relationships
    */
   public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
+    iteratee: ResourceIteratee<OktaUser>,
   ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
+    const users: OktaUser[] = [];
+    await this.oktaClient.listUsers().each((e) => {
+      users.push(e);
+    });
 
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
+    this.usersList = users; //for use later in groups
```

Comment:
Cool. I didn't know about `jobState.iterateEntities()`. That would have been a good solution here. This line actually never got used and got deleted in a later commit, because I just restructured things to pull groups first and then users, so users can add themselves to groups (since the Okta API doesn't provide users from groups - it provides the groups a user belongs to as part of the user).